src/main/java/com/minecolonies/coremod/commands/colonycommands/ColonyTeleportCommand.java

```diff
@@ -56,14 +73,73 @@ public boolean canRankUseCommand(@NotNull final Colony colony, @NotNull final En
         return colony.getPermissions().hasPermission(player, Action.TELEPORT_TO_COLONY);
     }
 
+    @Override
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final ActionMenu actionMenu) throws CommandException
+    {
+        Colony colony = actionMenu.getColonyForArgument("colony");
+        if (null == colony)
+        {
+            final EntityPlayer player = actionMenu.getPlayerForArgument("player");
+            if (player != null)
+            {
+                IColony iColony = ColonyManager.getIColonyByOwner(server.getEntityWorld(), player);
+                if (null == iColony)
+                {
+                    if (sender instanceof EntityPlayer)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1066)

Reply:
this